UPGRADING.md

```diff
@@ -0,0 +1,169 @@
+# Microsoft Graph PHP SDK Upgrade Guide
+
+This guide highlights breaking changes, bug fixes and new features introduced during major upgrades.
+
+
+# Upgrading to 2.0.0-RC5
+- [New Features](#new-features)
+- [Breaking Changes](#breaking-changes)
+- [Bug Fixes](#bug-fixes)
+
+
+# New Features
+
+## Authentication Provider support
+Version 2 introduces an Authentication Provider that handles the fetching, caching and refreshing of tokens ensuring all your requests are always authenticated.
+
+The Authentication Provider provided wraps around PHP League's [OAuth 2.0 client](https://oauth2-client.thephpleague.com/) hence its name - `PhpLeagueAuthenticationProvider`. The `PhpLeagueAuthenticationProvider` currently supports the `client_credentials`, `authorization_code` and `on_behalf_of` OAuth 2.0 flows. It also supports using certificate-based client authentication alongside token-based client authentication.
```

Comment:
I think "token-based client authentication" should be "secret-based client authentication".

Reply:
Thanks for catching this. Updated in my latest commit.